controller/client/client.go

```diff
@@ -60,6 +62,21 @@ func NewClientWithHTTP(uri, key string, httpClient *http.Client) (*Client, error
 
 // NewClientWithPin acts like NewClient, but specifies a TLS pin.
 func NewClientWithPin(uri, key string, pin []byte) (*Client, error) {
+	return NewClientWithPinConfig(uri, key, &pinned.Config{Pin: pin})
+}
+
+// NewClientWithPinDomain acts like NewClient, but specifies a TLS pin and
+// custom domain (useful for when the route does not resolve, e.g. in tests).
+func NewClientWithPinDomain(uri, key string, pin []byte, domain string) (*Client, error) {
```

Comment:
I wonder if we should just have a client config struct instead of the less common builder funcs?